software/dimos/src/io/can.cpp

```diff
@@ -0,0 +1,62 @@
+#include "can.h"
+#include <cstring>
+#include <unistd.h>
+#include <net/if.h>
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+#include <linux/can.h>
+
+std::optional<int> CanInterface;
```

Comment:
best practice would be to mark this static, so it can't be accidentally `extern`ed